src/TicketNow/TicketNow.Service/Services/OrderService.cs

```diff
@@ -0,0 +1,176 @@
+﻿using AutoMapper;
+using Microsoft.Extensions.Configuration;
+using RestSharp;
+using TicketNow.Domain.Dtos.Default;
+using TicketNow.Domain.Dtos.MockPayment;
+using TicketNow.Domain.Dtos.Order;
+using TicketNow.Domain.Entities;
+using TicketNow.Domain.Enums;
+using TicketNow.Domain.Interfaces.Repositories;
+using TicketNow.Domain.Interfaces.Services;
+using TicketNow.Infra.CrossCutting.Notifications;
+using TicketNow.Service.Validators.Order;
+
+namespace TicketNow.Service.Services
+{
+    public class OrderService : BaseService, IOrderService
+    {
+        private readonly IOrderRepository _orderRepository;
+        private readonly IOrderItemRepository _orderItemRepository;
+        private readonly IEventRepository _eventRepository;
+        private readonly IMapper _mapper;
+        private readonly IConfiguration _configuration;
+        private readonly NotificationContext _notificationContext;
+        private static string _apiKey;
+
+        public OrderService(IOrderRepository orderRepository, IOrderItemRepository orderItemRepository, IMapper mapper, IConfiguration configuration,
+            NotificationContext notificationContext, IEventRepository eventRepository)
+        {
+            _orderRepository = orderRepository;
+            _mapper = mapper;
+            _orderItemRepository = orderItemRepository;
+            _configuration = configuration;
+            _notificationContext = notificationContext;
+            _eventRepository = eventRepository;
+            _apiKey = StaticMockPaymentApiKey.ApiKey;
+        }
+
+        public async Task<DefaultServiceResponseDto> InsertNewOrderAsync(OrderDto newOrderDto)
+        {
+            _apiKey = StaticMockPaymentApiKey.ApiKey;
+            var eventDb = _eventRepository.Select(newOrderDto.EventId);
+            newOrderDto.EventActive = eventDb.Active;
+            newOrderDto.TicketsAvaiable = eventDb.TicketAvailable;
+
+            var validationResult = Validate(newOrderDto, Activator.CreateInstance<AddOrderValidator>());
+            if (!validationResult.IsValid) 
+            {
+                _notificationContext.AddNotifications(validationResult.Errors);
+                return default(DefaultServiceResponseDto);
+            }
+            
+            var orderDb = _mapper.Map<Order>(newOrderDto);
+            orderDb.CreatedAt = DateTime.Now;
+            orderDb.Status = OrderStatusEnum.Active;
+            orderDb.PaymentStatus = PaymentStatusEnum.WaitingPayment;
+            orderDb.Price = eventDb.TicketPrice * newOrderDto.Tickets;
+
+            var newOrderDb = await _orderRepository.InsertWithReturnId(orderDb);
+
+            for (var ticket = 0; ticket < newOrderDto.Tickets; ticket++)
+            {
```

Comment:
como vai receber esse cara por parametro, como uma lista dentro do AddOrderDto, nao precisa desse for, só precisa preencher o orderId nelas